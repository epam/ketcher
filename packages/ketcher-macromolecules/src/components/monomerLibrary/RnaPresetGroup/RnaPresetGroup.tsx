/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import { useAppSelector } from 'hooks';
import { MouseEvent } from 'react';
import {
  selectActivePreset,
  setActivePreset,
  setActivePresetForContextMenu,
  setIsEditMode,
} from 'state/rna-builder';
import { useDispatch } from 'react-redux';
import { RnaPresetItem } from 'components/monomerLibrary/RnaPresetItem';
import {
  GroupContainer,
  ItemsContainer,
} from 'components/monomerLibrary/monomerLibraryGroup/styles';
import { selectEditor } from 'state/common';
import { RNAContextMenu } from 'components/contextMenu/RNAContextMenu';
import { CONTEXT_MENU_ID } from 'components/contextMenu/types';
import { useContextMenu } from 'react-contexify';
import { IRnaPreset } from '../RnaBuilder/types';

export const RnaPresetGroup = ({ presets, duplicatePreset, editPreset }) => {
  const activePreset = useAppSelector(selectActivePreset);
  const editor = useAppSelector(selectEditor);

  const { show } = useContextMenu({ id: CONTEXT_MENU_ID.FOR_RNA });

  const dispatch = useDispatch();

  const selectPreset = (preset: IRnaPreset) => () => {
    dispatch(setActivePreset(preset));
    editor.events.selectPreset.dispatch(preset);
    if (preset === activePreset?.presetInList) return;
    dispatch(setIsEditMode(false));
  };

  const getMenuPosition = (currentPresetCard: HTMLElement) => {
    const isDivCard = currentPresetCard instanceof HTMLDivElement;
    if (!isDivCard && currentPresetCard.parentElement) {
      currentPresetCard = currentPresetCard.parentElement;
    }
    const boundingBox = currentPresetCard.getBoundingClientRect();
    const parentBox = currentPresetCard.offsetParent?.getBoundingClientRect();
    const contextMenuWidth = 140;
    let x = boundingBox.right - contextMenuWidth;
    const y = boundingBox.y + boundingBox.height / 2;
    if (parentBox?.x && parentBox?.x > x) {
      x = boundingBox.x;
    }
    return {
      x,
      y,
    };
  };

  const handleContextMenu = (preset: IRnaPreset) => (event: MouseEvent) => {
    event.stopPropagation();
    dispatch(setActivePresetForContextMenu(preset));
    show({
      event,
      props: {
        duplicatePreset,
        editPreset,
      },
      position: getMenuPosition(event.currentTarget as HTMLElement),
    });
  };

  return (
    <GroupContainer data-testid="rna-preset-group">
      <ItemsContainer>
        {presets.map((preset, index) => {
          return (
            <RnaPresetItem
              key={`${preset.name}${index}`}
              preset={preset}
              onClick={selectPreset(preset)}
              onContextMenu={handleContextMenu(preset)}
              isSelected={activePreset?.name === preset.name}
            />
          );
        })}
      </ItemsContainer>
      <RNAContextMenu />
    </GroupContainer>
  );
};
