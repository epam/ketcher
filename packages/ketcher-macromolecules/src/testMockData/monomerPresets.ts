import {
  AttachmentPointName,
  KetConnectionType,
  KetMonomerClass,
  KetMonomerGroupTemplateClass,
  KetTemplateType,
  MonomerItemType,
  Struct,
} from 'ketcher-core';

// you can add other props in monomers, if you need them
export const monomers: MonomerItemType[] = [
  {
    props: {
      MonomerName: '12ddR',
      Name: "1',2'-Di-Deoxy-Ribose",
      MonomerNaturalAnalogCode: '',
      MonomerType: '',
      BranchMonomer: '',
      MonomerCaps: {},
      MonomerCode: '',
    },
    label: '',
    struct: new Struct(),
  },
  {
    props: {
      MonomerName: '3A6',
      Name: "6-amino-hexanol (3' end)",
      MonomerNaturalAnalogCode: '',
      MonomerType: '',
      BranchMonomer: '',
      MonomerCaps: {},
      MonomerCode: '',
    },
    label: '',
    struct: new Struct(),
  },
  {
    props: {
      MonomerName: '3FAM',
      Name: '3-FAM',
      MonomerNaturalAnalogCode: '',
      MonomerType: '',
      BranchMonomer: '',
      MonomerCaps: {},
      MonomerCode: '',
    },
    label: '',
    struct: new Struct(),
  },
  {
    props: {
      MonomerName: '4sR',
      Name: '4-Thio-Ribose',
      MonomerNaturalAnalogCode: '',
      MonomerType: '',
      BranchMonomer: '',
      MonomerCaps: {},
      MonomerCode: '',
    },
    label: '',
    struct: new Struct(),
  },
];

export const preset = {
  base: {
    struct: {
      atoms: {
        nextId: 9,
      },
      bonds: {
        nextId: 9,
      },
      sgroups: {
        nextId: 0,
      },
      halfBonds: {
        nextId: 0,
      },
      loops: {
        nextId: 0,
      },
      isReaction: false,
      rxnArrows: {
        nextId: 0,
      },
      rxnPluses: {
        nextId: 0,
      },
      frags: {
        nextId: 1,
      },
      rgroups: {
        nextId: 0,
      },
      rgroupAttachmentPoints: {
        nextId: 0,
      },
      name: 'Uracil',
      abbreviation: '',
      sGroupForest: {
        parent: {},
        children: {},
        atomSets: {},
      },
      simpleObjects: {
        nextId: 0,
      },
      texts: {
        nextId: 0,
      },
      functionalGroups: {
        nextId: 0,
      },
      highlights: {
        nextId: 0,
      },
    },
    props: {
      Name: 'Uracil',
      MonomerType: 'RNA',
      MonomerName: 'U',
      MonomerCode: 'U',
      MonomerNaturalAnalogCode: 'U',
      BranchMonomer: 'true',
      MonomerCaps: {
        R1: 'H',
      },
    },
    favorite: false,
    label: 'U',
  },
  sugar: {
    struct: {
      atoms: {
        nextId: 12,
      },
      bonds: {
        nextId: 12,
      },
      sgroups: {
        nextId: 0,
      },
      halfBonds: {
        nextId: 0,
      },
      loops: {
        nextId: 0,
      },
      isReaction: false,
      rxnArrows: {
        nextId: 0,
      },
      rxnPluses: {
        nextId: 0,
      },
      frags: {
        nextId: 1,
      },
      rgroups: {
        nextId: 0,
      },
      rgroupAttachmentPoints: {
        nextId: 0,
      },
      name: 'Ribose',
      abbreviation: '',
      sGroupForest: {
        parent: {},
        children: {},
        atomSets: {},
      },
      simpleObjects: {
        nextId: 0,
      },
      texts: {
        nextId: 0,
      },
      functionalGroups: {
        nextId: 0,
      },
      highlights: {
        nextId: 0,
      },
    },
    props: {
      Name: 'Ribose',
      MonomerType: 'RNA',
      MonomerName: 'R',
      MonomerCode: 'R',
      MonomerNaturalAnalogCode: 'R',
      BranchMonomer: 'false',
      MonomerCaps: {
        R1: 'H',
        R2: 'H',
        R3: 'O',
      },
    },
    favorite: false,
    label: 'R',
  },
  phosphate: {
    struct: {
      atoms: {
        nextId: 5,
      },
      bonds: {
        nextId: 4,
      },
      sgroups: {
        nextId: 0,
      },
      halfBonds: {
        nextId: 0,
      },
      loops: {
        nextId: 0,
      },
      isReaction: false,
      rxnArrows: {
        nextId: 0,
      },
      rxnPluses: {
        nextId: 0,
      },
      frags: {
        nextId: 1,
      },
      rgroups: {
        nextId: 0,
      },
      rgroupAttachmentPoints: {
        nextId: 0,
      },
      name: 'Phosphate',
      abbreviation: '',
      sGroupForest: {
        parent: {},
        children: {},
        atomSets: {},
      },
      simpleObjects: {
        nextId: 0,
      },
      texts: {
        nextId: 0,
      },
      functionalGroups: {
        nextId: 0,
      },
      highlights: {
        nextId: 0,
      },
    },
    props: {
      Name: 'Phosphate',
      MonomerType: 'RNA',
      MonomerName: 'P',
      MonomerCode: 'P',
      MonomerNaturalAnalogCode: 'P',
      BranchMonomer: 'false',
      MonomerCaps: {
        R1: 'O',
        R2: 'O',
      },
    },
    favorite: false,
    label: 'P',
  },
  name: 'U',
  default: true,
};

const phosphate = {
  props: {
    MonomerName: 'P',
    Name: 'Phosphate',
    MonomerNaturalAnalogCode: 'P',
    MonomerType: '',
    MonomerClass: KetMonomerClass.Phosphate,
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: 'P',
  struct: new Struct(),
};
const ribose = {
  props: {
    MonomerName: 'R',
    Name: 'Ribose',
    MonomerNaturalAnalogCode: 'R',
    MonomerType: '',
    MonomerClass: KetMonomerClass.Sugar,
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: 'R',
  struct: new Struct(),
};
const thymine = {
  props: {
    MonomerName: 'T',
    Name: 'Thymine',
    MonomerNaturalAnalogCode: 'T',
    MonomerType: '',
    MonomerClass: KetMonomerClass.Base,
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: 'T',
  struct: new Struct(),
};
const cytosine = {
  props: {
    MonomerName: 'C',
    Name: 'Cytosine',
    MonomerNaturalAnalogCode: 'C',
    MonomerType: '',
    MonomerClass: KetMonomerClass.Base,
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: '',
  struct: new Struct(),
};
const uracil = {
  props: {
    MonomerName: 'U',
    Name: 'Uracil',
    MonomerNaturalAnalogCode: 'U',
    MonomerType: '',
    MonomerClass: KetMonomerClass.Base,
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: '',
  struct: new Struct(),
};
const adenine = {
  props: {
    MonomerName: 'A',
    Name: 'Adenine',
    MonomerNaturalAnalogCode: 'A',
    MonomerType: '',
    MonomerClass: KetMonomerClass.Base,
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: '',
  struct: new Struct(),
};
const guanine = {
  props: {
    MonomerName: 'G',
    Name: 'Guanine',
    MonomerNaturalAnalogCode: 'G',
    MonomerType: '',
    MonomerClass: KetMonomerClass.Base,
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: 'G',
  struct: new Struct(),
};

const rnaPresetsTemplates = [
  {
    type: KetTemplateType.MONOMER_GROUP_TEMPLATE,
    id: 'A',
    name: 'A',
    class: KetMonomerGroupTemplateClass.RNA,
    templates: [
      {
        $ref: 'monomerTemplate-R___Ribose',
      },
      {
        $ref: 'monomerTemplate-A___Adenine',
      },
      {
        $ref: 'monomerTemplate-P___Phosphate',
      },
    ],
    connections: [
      {
        connectionType: KetConnectionType.SINGLE,
        endpoint1: {
          monomerTemplateId: 'monomerTemplate-A___Adenine',
          attachmentPointId: AttachmentPointName.R1,
        },
        endpoint2: {
          monomerTemplateId: 'monomerTemplate-R___Ribose',
          attachmentPointId: AttachmentPointName.R3,
        },
      },
      {
        connectionType: KetConnectionType.SINGLE,
        endpoint1: {
          monomerTemplateId: 'monomerTemplate-R___Ribose',
          attachmentPointId: AttachmentPointName.R2,
        },
        endpoint2: {
          monomerTemplateId: 'monomerTemplate-P___Phosphate',
          attachmentPointId: AttachmentPointName.R1,
        },
      },
    ],
  },
  {
    type: KetTemplateType.MONOMER_GROUP_TEMPLATE,
    id: 'C',
    name: 'C',
    class: KetMonomerGroupTemplateClass.RNA,
    templates: [
      {
        $ref: 'monomerTemplate-R___Ribose',
      },
      {
        $ref: 'monomerTemplate-C___Cytosine',
      },
      {
        $ref: 'monomerTemplate-P___Phosphate',
      },
    ],
    connections: [
      {
        connectionType: KetConnectionType.SINGLE,
        endpoint1: {
          monomerTemplateId: 'monomerTemplate-C___Cytosine',
          attachmentPointId: AttachmentPointName.R1,
        },
        endpoint2: {
          monomerTemplateId: 'monomerTemplate-R___Ribose',
          attachmentPointId: AttachmentPointName.R3,
        },
      },
      {
        connectionType: KetConnectionType.SINGLE,
        endpoint1: {
          monomerTemplateId: 'monomerTemplate-R___Ribose',
          attachmentPointId: AttachmentPointName.R2,
        },
        endpoint2: {
          monomerTemplateId: 'monomerTemplate-P___Phosphate',
          attachmentPointId: AttachmentPointName.R1,
        },
      },
    ],
  },
  {
    type: KetTemplateType.MONOMER_GROUP_TEMPLATE,
    id: 'G',
    name: 'G',
    class: KetMonomerGroupTemplateClass.RNA,
    templates: [
      {
        $ref: 'monomerTemplate-R___Ribose',
      },
      {
        $ref: 'monomerTemplate-G___Guanine',
      },
      {
        $ref: 'monomerTemplate-P___Phosphate',
      },
    ],
    connections: [
      {
        connectionType: KetConnectionType.SINGLE,
        endpoint1: {
          monomerTemplateId: 'monomerTemplate-G___Guanine',
          attachmentPointId: AttachmentPointName.R1,
        },
        endpoint2: {
          monomerTemplateId: 'monomerTemplate-R___Ribose',
          attachmentPointId: AttachmentPointName.R3,
        },
      },
      {
        connectionType: KetConnectionType.SINGLE,
        endpoint1: {
          monomerTemplateId: 'monomerTemplate-R___Ribose',
          attachmentPointId: AttachmentPointName.R2,
        },
        endpoint2: {
          monomerTemplateId: 'monomerTemplate-P___Phosphate',
          attachmentPointId: AttachmentPointName.R1,
        },
      },
    ],
  },
  {
    type: KetTemplateType.MONOMER_GROUP_TEMPLATE,
    id: 'T',
    name: 'T',
    class: KetMonomerGroupTemplateClass.RNA,
    templates: [
      {
        $ref: 'monomerTemplate-R___Ribose',
      },
      {
        $ref: 'monomerTemplate-T___Thymine',
      },
      {
        $ref: 'monomerTemplate-P___Phosphate',
      },
    ],
    connections: [
      {
        connectionType: KetConnectionType.SINGLE,
        endpoint1: {
          monomerTemplateId: 'monomerTemplate-T___Thymine',
          attachmentPointId: AttachmentPointName.R1,
        },
        endpoint2: {
          monomerTemplateId: 'monomerTemplate-R___Ribose',
          attachmentPointId: AttachmentPointName.R3,
        },
      },
      {
        connectionType: KetConnectionType.SINGLE,
        endpoint1: {
          monomerTemplateId: 'monomerTemplate-R___Ribose',
          attachmentPointId: AttachmentPointName.R2,
        },
        endpoint2: {
          monomerTemplateId: 'monomerTemplate-P___Phosphate',
          attachmentPointId: AttachmentPointName.R1,
        },
      },
    ],
  },
  {
    type: KetTemplateType.MONOMER_GROUP_TEMPLATE,
    id: 'U',
    name: 'U',
    class: KetMonomerGroupTemplateClass.RNA,
    templates: [
      {
        $ref: 'monomerTemplate-R___Ribose',
      },
      {
        $ref: 'monomerTemplate-U___Uracil',
      },
      {
        $ref: 'monomerTemplate-P___Phosphate',
      },
    ],
    connections: [
      {
        connectionType: KetConnectionType.SINGLE,
        endpoint1: {
          monomerTemplateId: 'monomerTemplate-U___Uracil',
          attachmentPointId: AttachmentPointName.R1,
        },
        endpoint2: {
          monomerTemplateId: 'monomerTemplate-R___Ribose',
          attachmentPointId: AttachmentPointName.R3,
        },
      },
      {
        connectionType: KetConnectionType.SINGLE,
        endpoint1: {
          monomerTemplateId: 'monomerTemplate-R___Ribose',
          attachmentPointId: AttachmentPointName.R2,
        },
        endpoint2: {
          monomerTemplateId: 'monomerTemplate-P___Phosphate',
          attachmentPointId: AttachmentPointName.R1,
        },
      },
    ],
  },
];

export {
  phosphate,
  ribose,
  cytosine,
  guanine,
  thymine,
  uracil,
  adenine,
  rnaPresetsTemplates,
};
