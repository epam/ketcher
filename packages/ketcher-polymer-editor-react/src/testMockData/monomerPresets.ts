import { MonomerItemType, Struct } from 'ketcher-core';

// you can add other props in monomers, if you need them
export const monomers: MonomerItemType[] = [
  {
    props: {
      MonomerName: '12ddR',
      Name: "1',2'-Di-Deoxy-Ribose",
      MonomerNaturalAnalogCode: '',
      MonomerType: '',
      BranchMonomer: '',
      MonomerCaps: {},
      MonomerCode: '',
    },
    label: '',
    struct: new Struct(),
  },
  {
    props: {
      MonomerName: '3A6',
      Name: "6-amino-hexanol (3' end)",
      MonomerNaturalAnalogCode: '',
      MonomerType: '',
      BranchMonomer: '',
      MonomerCaps: {},
      MonomerCode: '',
    },
    label: '',
    struct: new Struct(),
  },
  {
    props: {
      MonomerName: '3FAM',
      Name: '3-FAM',
      MonomerNaturalAnalogCode: '',
      MonomerType: '',
      BranchMonomer: '',
      MonomerCaps: {},
      MonomerCode: '',
    },
    label: '',
    struct: new Struct(),
  },
  {
    props: {
      MonomerName: '4sR',
      Name: '4-Thio-Ribose',
      MonomerNaturalAnalogCode: '',
      MonomerType: '',
      BranchMonomer: '',
      MonomerCaps: {},
      MonomerCode: '',
    },
    label: '',
    struct: new Struct(),
  },
];

export const preset = {
  base: {
    struct: {
      atoms: {
        nextId: 9,
      },
      bonds: {
        nextId: 9,
      },
      sgroups: {
        nextId: 0,
      },
      halfBonds: {
        nextId: 0,
      },
      loops: {
        nextId: 0,
      },
      isReaction: false,
      rxnArrows: {
        nextId: 0,
      },
      rxnPluses: {
        nextId: 0,
      },
      frags: {
        nextId: 1,
      },
      rgroups: {
        nextId: 0,
      },
      rgroupAttachmentPoints: {
        nextId: 0,
      },
      name: 'Uracil',
      abbreviation: '',
      sGroupForest: {
        parent: {},
        children: {},
        atomSets: {},
      },
      simpleObjects: {
        nextId: 0,
      },
      texts: {
        nextId: 0,
      },
      functionalGroups: {
        nextId: 0,
      },
      highlights: {
        nextId: 0,
      },
    },
    props: {
      Name: 'Uracil',
      MonomerType: 'RNA',
      MonomerName: 'U',
      MonomerCode: 'U',
      MonomerNaturalAnalogCode: 'U',
      BranchMonomer: 'true',
      MonomerCaps: {
        R1: 'H',
      },
    },
    favorite: false,
    label: 'U',
  },
  sugar: {
    struct: {
      atoms: {
        nextId: 12,
      },
      bonds: {
        nextId: 12,
      },
      sgroups: {
        nextId: 0,
      },
      halfBonds: {
        nextId: 0,
      },
      loops: {
        nextId: 0,
      },
      isReaction: false,
      rxnArrows: {
        nextId: 0,
      },
      rxnPluses: {
        nextId: 0,
      },
      frags: {
        nextId: 1,
      },
      rgroups: {
        nextId: 0,
      },
      rgroupAttachmentPoints: {
        nextId: 0,
      },
      name: 'Ribose',
      abbreviation: '',
      sGroupForest: {
        parent: {},
        children: {},
        atomSets: {},
      },
      simpleObjects: {
        nextId: 0,
      },
      texts: {
        nextId: 0,
      },
      functionalGroups: {
        nextId: 0,
      },
      highlights: {
        nextId: 0,
      },
    },
    props: {
      Name: 'Ribose',
      MonomerType: 'RNA',
      MonomerName: 'R',
      MonomerCode: 'R',
      MonomerNaturalAnalogCode: 'R',
      BranchMonomer: 'false',
      MonomerCaps: {
        R1: 'H',
        R2: 'H',
        R3: 'O',
      },
    },
    favorite: false,
    label: 'R',
  },
  phosphate: {
    struct: {
      atoms: {
        nextId: 5,
      },
      bonds: {
        nextId: 4,
      },
      sgroups: {
        nextId: 0,
      },
      halfBonds: {
        nextId: 0,
      },
      loops: {
        nextId: 0,
      },
      isReaction: false,
      rxnArrows: {
        nextId: 0,
      },
      rxnPluses: {
        nextId: 0,
      },
      frags: {
        nextId: 1,
      },
      rgroups: {
        nextId: 0,
      },
      rgroupAttachmentPoints: {
        nextId: 0,
      },
      name: 'Phosphate',
      abbreviation: '',
      sGroupForest: {
        parent: {},
        children: {},
        atomSets: {},
      },
      simpleObjects: {
        nextId: 0,
      },
      texts: {
        nextId: 0,
      },
      functionalGroups: {
        nextId: 0,
      },
      highlights: {
        nextId: 0,
      },
    },
    props: {
      Name: 'Phosphate',
      MonomerType: 'RNA',
      MonomerName: 'P',
      MonomerCode: 'P',
      MonomerNaturalAnalogCode: 'P',
      BranchMonomer: 'false',
      MonomerCaps: {
        R1: 'O',
        R2: 'O',
      },
    },
    favorite: false,
    label: 'P',
  },
  name: 'U',
  default: true,
};

const phosphate = {
  props: {
    MonomerName: 'P',
    Name: 'Phosphate',
    MonomerNaturalAnalogCode: '',
    MonomerType: '',
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: 'P',
  struct: new Struct(),
};
const ribose = {
  props: {
    MonomerName: 'R',
    Name: 'Ribose',
    MonomerNaturalAnalogCode: '',
    MonomerType: '',
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: 'R',
  struct: new Struct(),
};
const thymine = {
  props: {
    MonomerName: 'T',
    Name: 'Thymine',
    MonomerNaturalAnalogCode: '',
    MonomerType: '',
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: 'T',
  struct: new Struct(),
};
const cytosine = {
  props: {
    MonomerName: 'C',
    Name: 'Cytosine',
    MonomerNaturalAnalogCode: '',
    MonomerType: '',
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: '',
  struct: new Struct(),
};
const uracil = {
  props: {
    MonomerName: 'U',
    Name: 'Uracil',
    MonomerNaturalAnalogCode: '',
    MonomerType: '',
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: '',
  struct: new Struct(),
};
const adenine = {
  props: {
    MonomerName: 'A',
    Name: 'Adenine',
    MonomerNaturalAnalogCode: '',
    MonomerType: '',
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: '',
  struct: new Struct(),
};
const guanine = {
  props: {
    MonomerName: 'G',
    Name: 'Guanine',
    MonomerNaturalAnalogCode: '',
    MonomerType: '',
    BranchMonomer: '',
    MonomerCaps: {},
    MonomerCode: '',
  },
  label: 'G',
  struct: new Struct(),
};

export { phosphate, ribose, cytosine, guanine, thymine, uracil, adenine };
