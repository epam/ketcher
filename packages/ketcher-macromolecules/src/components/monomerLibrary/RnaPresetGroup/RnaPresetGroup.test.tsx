import { render, screen } from '@testing-library/react';
import { RnaPresetGroup } from './RnaPresetGroup';
import { preset } from 'src/testMockData/monomerPresets';

const duplicatePreset = jest.fn();
const editPreset = jest.fn();
describe('Test Rna Preset Group component', () => {
  it('should render', () => {
    render(
      withThemeAndStoreProvider(
        <RnaPresetGroup
          presets={[preset]}
          duplicatePreset={duplicatePreset}
          editPreset={editPreset}
        />,
      ),
    );

    const rnaPresetGroup = screen.getByTestId('rna-preset-group');

    expect(rnaPresetGroup).toMatchSnapshot();
  });
});
