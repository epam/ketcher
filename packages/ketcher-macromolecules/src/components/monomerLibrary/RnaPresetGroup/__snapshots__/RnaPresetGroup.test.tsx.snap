// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rna Preset Group component should render 1`] = `
<div
  class="css-1jn33hb"
  data-testid="rna-preset-group"
>
  <div
    class="css-14bnc38"
  >
    <div
      class="css-18vlgix"
      data-rna-preset-item-name="U"
      data-testid="U_U_R_P"
    >
      <span>
        U
      </span>
      <svg
        class="dots hidden css-1qr96ji"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="8"
          cy="3.5"
          fill="#7C7C7F"
          r="1.5"
        />
        <circle
          cx="8"
          cy="8"
          fill="#7C7C7F"
          r="1.5"
        />
        <circle
          cx="8"
          cy="12.5"
          fill="#7C7C7F"
          r="1.5"
        />
      </svg>
      <div
        aria-hidden="true"
        class="star "
      >
        â˜…
      </div>
    </div>
  </div>
</div>
`;
